dev-docs/RFCs/v6.x/gpu-aggregation-enhancements.md

```diff
@@ -0,0 +1,112 @@
+
+Multiple weights can be supported by internally maintaining a separate texture (used as render target) for each weights array. For each weights array, we set required blend modes based on requested aggregation operation and perform render calls and collect aggregated data into corresponding texture. We could use `Multiple Render Target` feature to perform single draw for aggregating multiple weights, but WebGL2 **does not** support blending per render target, blend setup is global and applies to all render targets. Users can still use `size` parameter, as described below, to pack more than one weight (up to 3) and aggregate all of them with single draw call, provided, aggregation operation is same for all of them.
+
+
+### Update `run` method
+
+The public API that needs to change is `run()` method. We change input arguments and return object contents. Current style can still be supported as deprecated.
+
+#### weights arguments
+
+Update `run` method to accept weights as an array of objects with following values. Each object will result in one aggregation cycle.
+
+  * id (String) : used to identify the corresponding aggregation buffer and texture in return object. Must be unique for each weights object.
+  * values (Array) : ArrayBufferView with weight values.
+  * size (Number, default: 1, minValue: 1, maxValue: 3) : Size of a single weight instance. Determines how many distinct weights exist in the array.
+  * operation (Enum {SUM, MEAN, MIN or MAX}, default: SUM) : Defines aggregation operation.
+  * needMinMax (Boolean, default: false) : when true additional aggregation steps are performed to calculate total min and max aggregation values.
+  * combineMinMax (Boolean, default: false) : Applicable only when `needMinMax` is set. When true, both min and max values are calculated in single aggregation step, but only does this for first weight (i.e if `size` is 2 or 3, min and max values are calculated only for first weight)
+
+Existing single weight array can be supported as deprecated by internally converting it into an array with single object, {id: 'weights', values: weights, needMinMax: true, combineMinMax: true}.
+
+Other than `weights` all other arguments to `run` remain same.
+
+#### return object
+
+Result of `run` is an array of objects with one object corresponds to an object in `weights` array parameter. Each object contains following values
+
+  * aggregationBuffer (Buffer) : Aggregated values per grid cell, aggregation is performed as per specified `operation`. Size of the buffer is 4, with R, G and B channel corresponds to aggregated weights. When input `size` is < 3, G or B channels contain undefined values. A channel contains count of points aggregated into this cell.
+
+  * aggregationTexture (Texture, optional) : When aggregation is performed on GPU, contains above data as form of texture, useful for applications that want to consume the a texture instead of buffer. This value is `null` when aggregation is performed on CPU.
```

Comment:
Is it either buffer or texture? the latter is listed as optional but not the first?

Do we need to read out the buffer if someone wants to consume the texture?

Reply:
Updated the text, `Buffers` are always present and gives unique API for both CPU and GPU aggregations. `Textures` are returned when aggregation is done on GPU, they will be set to `null` when aggregation is done on CPU.